rust/src/ldap/detect.rs

```diff
@@ -318,6 +320,65 @@ unsafe extern "C" fn ldap_tx_get_request_dn(
     return false;
 }
 
+unsafe extern "C" fn ldap_detect_responses_dn_setup(
+    de: *mut c_void, s: *mut c_void, _raw: *const std::os::raw::c_char,
+) -> c_int {
+    if DetectSignatureSetAppProto(s, ALPROTO_LDAP) != 0 {
+        return -1;
+    }
+    if DetectBufferSetActiveList(de, s, G_LDAP_RESPONSES_DN_BUFFER_ID) < 0 {
+        return -1;
+    }
+    return 0;
+}
+
+unsafe extern "C" fn ldap_detect_responses_dn_get_data(
+    de: *mut c_void, transforms: *const c_void, flow: *const c_void, flow_flags: u8,
+    tx: *const c_void, list_id: c_int, local_id: u32,
+) -> *mut c_void {
+    return DetectHelperGetMultiData(
+        de,
+        transforms,
+        flow,
+        flow_flags,
+        tx,
+        list_id,
+        local_id,
+        ldap_tx_get_responses_dn,
+    );
+}
+
+unsafe extern "C" fn ldap_tx_get_responses_dn(
+    tx: *const c_void, _flags: u8, local_id: u32, buffer: *mut *const u8, buffer_len: *mut u32,
+) -> bool {
+    let tx = cast_pointer!(tx, LdapTransaction);
+
+    if local_id as usize >= tx.responses.len() {
+        return false;
+    }
+    *buffer = std::ptr::null();
+    *buffer_len = 0;
+
+    let response = &tx.responses[local_id as usize];
+    // We expect every response in one tx to be the same protocol_op
```

Comment:
What about IntermediateResponse and SearchResultReference cases ?

Reply:
For Intermediate Response, we should find real pcaps with it to see how it happens

Reply:
And we also need a pcap with SearchResultReference

Reply:
(For me, the PR is good, but I would like to better test/investigate/understand these cases)